### ISSUE TITLE
Custom type validator could not be found in an inherited class (from Validator)

### ISSUE THREAD

**Issue by anatoliis** [pyeve/cerberus#336]
According to the docs, this code should work, but it doesn't.
### What I have done:
- Created a custom field validator `intrange`
- Created a custom type validator `intrange`
- Added a docstring to the `_validate_intrange` method, declaring the expected type of `intrange` parameter
### What I expect:
- I expect the value `range(2, 10)` to be passed to `_validate_type_intrange` method
- If the previous method returns `True`, this value should be passed to the `_validate_intrange` method along with `field` and `value` parameters

```python
from cerberus import Validator


class DataValidator(Validator):

    def _validate_intrange(self, intrange, field, value):
        """ The rule's arguments are validated against this schema:
        {'type': 'intrange'}
        """
        if value not in intrange:
            self._error(
                field,
                f'must be in the range of <{min(intrange)}, {max(intrange)}>'
            )

    def _validate_type_intrange(self, value):
        if isinstance(value, range):
            return True


SCHEMA = {
    'first': {
        'required': True,
        'type': 'integer',
        'intrange': range(2, 10)
    }
}


validator = DataValidator(SCHEMA)
```
Using validator:
```python
data = {
    'first': 3
}

if not validator.validate(data):
    print('errors:', validator.errors)
else:
    print('OK')
```

### But I'm getting an error instead:
```
~/python-envs/some_env/bin/python3.6 ~/PycharmProjects/cerberus_test/main.py
Traceback (most recent call last):
  File "~/PycharmProjects/cerberus_test/main.py", line 1, in <module>
    from validator import validator
  File "~/PycharmProjects/cerberus_test/validator.py", line 27, in <module>
    validator = DataValidator(SCHEMA)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 143, in __init__
    self.schema = kwargs.get('schema', None)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 472, in schema
    self._schema = DefinitionSchema(self, schema)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/schema.py", line 56, in __init__
    self.validate(schema)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/schema.py", line 181, in validate
    self._validate(schema)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/schema.py", line 202, in _validate
    if not self.schema_validator(schema, normalize=False):
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 805, in validate
    self.__validate_unknown_fields(field)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 838, in __validate_unknown_fields
    if not validator({field: value}, normalize=False):
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 803, in validate
    self.__validate_definitions(definitions, field)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 871, in __validate_definitions
    validate_rule(rule)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 852, in validate_rule
    return validator(definitions.get(rule, None), field, value)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 1153, in _validate_schema
    self.__validate_schema_mapping(field, schema, value)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 1164, in __validate_schema_mapping
    if not validator(value, update=self.update, normalize=False):
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 803, in validate
    self.__validate_definitions(definitions, field)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 862, in __validate_definitions
    if validate_rule(rule):
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 852, in validate_rule
    return validator(definitions.get(rule, None), field, value)
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 1186, in _validate_type
    for x in types):
  File "~/python-envs/some_env/lib/python3.6/site-packages/cerberus/validator.py", line 1186, in <genexpr>
    for x in types):
TypeError: 'NoneType' object is not callable

Process finished with exit code 1
```
As it can be seen from the error traceback, the method `_validate_type_intrange` couldn't be found by cerberus.
`cerberus/validator.py", line 852, in validate_rule`:
`definitions.get(rule, None)` returns `None`

Is it a bug or maybe I'm missing something?

**Comment by funkyfuture** [comment]
does this also occur when you actually define `intrange` as expected type?

```python
SCHEMA = {
    'first': {
        'required': True,
        'type': 'intrange',
        'intrange': range(2, 10)
    }
}
```

not asking to ignore a possible bug, but why aren't `min` and `max` feasible?

**Comment by anatoliis** [comment]
@funkyfuture, yes, if expected type is `intrange`, I still keep getting (almost) the same error.

`min` and `max` are absolutely ok, but they're two separate validators with separate errors:
- `min value is 2`
- `max value is 3`
I just wanted to return an error, that contains minimum and maximum at the same time, e.g.:
- `f'must be in the range of <{min(intrange)}, {max(intrange)}>'`

**Comment by funkyfuture** [comment]
okay, it would be helpful if you provided:

- the full traceback
- a test case
- a bug fix :-)

i'm currently not capable of any work that requires concentration, so expect nothing from me in the foreseeable future.

this *might* also be the cause of #317/#318. when i looked into this more deeply, my suspicion was that there's an issue with child-validators not inheriting methods; `SchemaValidatorMixin._validator_schema` in this case. but i had no opportunity to follow this further.

**Comment by funkyfuture** [comment]
btw, i'd find a `range` rule that takes constraints like `'x-y'` appealing.

**Comment by funkyfuture** [comment]
> As it can be seen from the error traceback, the method `_validate_type_intrange` couldn't be found by cerberus.

there's not `intrange` given as constraint to `type` in your schema.

**Comment by aowolabi** [comment]
I am running into the same issue. I have a custom type method as well as a custom validator rule in a class that inherits from `Validator`.  When I attempt to use the custom type for rule level argument validation an exception occurs.

**Comment by funkyfuture** [comment]
@aowolabi have you checked there's no confusion between type-checking and rule-implementing methods  as with @anatoliis? yes, the customization should be less confusion prone, but we can't change this [within a major branch](https://github.com/pyeve/cerberus/pull/348/files#diff-38574c080d4e2eb38c49b86e6588ad98R61) to maintain backward-compatibility.

**Comment by funkyfuture** [comment]
this issue had actually been solved with #347, or not?
